### PR TITLE
Add buttons to toggle on and off the sampling profiler

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

**Comment by swankjesse** [comment]
Screen Shot 2022-02-08 at 17 31 29

Screen Shot 2022-02-08 at 17 31 55

File: samples/emoji-search/src/main/java/app/cash/zipline/samples/emojisearch/ProfilerToggle.kt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  private interface ProfilerState {
+    val action: String
+    fun toggle(): ProfilerState
+  }
+
+  private inner class Idle : ProfilerState {
+    override val action = "Start"
+
+    override fun toggle(): ProfilerState {
+      val now = SimpleDateFormat("yyyyMMdd-HHmmss", Locale.ROOT).format(Date())
```
Comment (JakeWharton) [inline samples/emoji-search/src/main/java/app/cash/zipline/samples/emojisearch/ProfilerToggle.kt]: No java.time?

File: samples/emoji-search/src/main/java/app/cash/zipline/samples/emojisearch/ProfilerToggle.kt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    }
+  }
+
+  private interface ProfilerState {
+    val action: String
+    fun toggle(): ProfilerState
+  }
+
+  private inner class Idle : ProfilerState {
+    override val action = "Start"
```
Comment (JakeWharton) [inline samples/emoji-search/src/main/java/app/cash/zipline/samples/emojisearch/ProfilerToggle.kt]:
```suggestion
    override val action get() = "Start"
```

File: samples/emoji-search/src/main/java/app/cash/zipline/samples/emojisearch/ProfilerToggle.kt [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      val out = File(baseDir, "emoji-search-$now.hprof")
+      val closeable = quickJs.startCpuSampling(out)
+      return Running(closeable)
+    }
+  }
+
+  private inner class Running(
+    private val closeable: Closeable
+  ) : ProfilerState {
+    override val action = "Stop"
```
Comment (JakeWharton) [inline samples/emoji-search/src/main/java/app/cash/zipline/samples/emojisearch/ProfilerToggle.kt]:
```suggestion
    override val action get() = "Stop"
```

**Review by JakeWharton — APPROVED** [review]
Feel free to merge as-is

File: samples/emoji-search/src/main/java/app/cash/zipline/samples/emojisearch/ProfilerToggle.kt (hunk 1 above)
Comment (swankjesse) [reply samples/emoji-search/src/main/java/app/cash/zipline/samples/emojisearch/ProfilerToggle.kt]: It wants API 26, I’m using API 21.

Comment (JakeWharton) [reply samples/emoji-search/src/main/java/app/cash/zipline/samples/emojisearch/ProfilerToggle.kt]: Desugar must not be turned on

**Comment by JakeWharton** [comment]
Now I feel bad for making you do that work. I said you could merge as-is!

**Pull request merged**